book/src/dev/rfcs/0008-release-planning.md

```diff
@@ -0,0 +1,191 @@
+additional work required to publish `zebrad` this way.  Second, we get a
+cross-platform update mechanism with no additional work, since `cargo install`
+will upgrade to the latest published version.  Third, we don't rely on any
+third parties to mediate the relationship between us and our users, so users
+can get updates as soon as we publish them.  Fourth, unlike a system package
+manager, we can pin exact hashes of every transitive dependency (via the
+`Cargo.lock`, which `cargo install` can be configured to respect).  Fifth,
+we're positioned to pick up (or contribute to) ecosystem-wide integrity
+improvements like a transparency log for `crates.io` or work on reproducible
+builds for Rust.
+
+This proposal is summarized above in the [guide-level
+explanation](#guide-level-explanation).
+
+## Release Processes
+
+The next question is what kind of release processes and automation we should
+use.  Here are two important priorities for these processes:
+
+1.  Reducing the friction of doing any individual release, allowing us to move
+    closer to a continuous deployment model;
+2.  Reducing the risk of error in the release process.
+
+These are roughly in order of priority but they're clearly related, since the
+more friction we have in the release process, the greater the risk of error,
+and the greater the risk of error, the more friction we require to prevent it.
+
+Automation helps to reduce friction and to reduce the risk of error, but
+although it helps enormously, it has declining returns after some point (in the
+sense that automating the final 5% of the work is significantly more complex
+and error-prone than automating the first 5% of the work).  So the challenge is
+to find the right "entry point" for the automated part of the system that
+strikes the right balance.
+
+One possibility is the following.  CD automation is triggered by pushing a new
+tag to the git repository.  Tags are specified as `crate-semver`, e.g.
+`zebra-network-3.2.8`, `zebrad-3.9.1`, etc.  When a new tag is pushed, the CD
+automation parses the tag to determine the crate name.  If it is `zebrad`, it
+builds a new Docker image and publishes the image and the crate.  Otherwise, it
+just publishes the crate.
+
+To publish a new version of any component crate, the process is:
+
+1.  Edit the `Cargo.toml` to increment the version number;
+2.  Update `crate/CHANGELOG.md` with a few human-readable sentences describing
+    changes since the last release (examples: [cl1], [cl2], [cl3]);
+3.  Submit a PR with these changes;
+4.  Tag the merge commit and push the tag to the git repo.
+
+[cl1]: https://github.com/ZcashFoundation/ed25519-zebra/blob/main/CHANGELOG.md
+[cl2]: https://github.com/dalek-cryptography/x25519-dalek/blob/master/CHANGELOG.md
+[cl3]: https://github.com/dalek-cryptography/curve25519-dalek/blob/master/CHANGELOG.md
+
+All subsequent steps (publishing to crates.io, building docker images, etc) are
+fully automated.
+
```

Comment:
Here's how we can automate building a release binary from a tag, using the exact dependencies in the lockfile:
```sh
$ git clone 'https://github.com/ZcashFoundation/zebra'
$ cd zebra
$ git checkout v1.0.0-alpha.0
$ cargo fetch
$ cargo build --frozen
```

See #1495 for some more details. (Most build scripts will work immediately after a release, but will fail as soon as a dependency updates. So we need to fetch dependencies and use `--frozen`.)